boilerplate/app/components/Icon.tsx

```diff
@@ -37,6 +37,11 @@ interface IconProps extends TouchableOpacityProps {
    * Style overrides for the icon container
    */
   containerStyle?: StyleProp<ViewStyle>
+
+  /**
+   * An optional function to be called when the icon is pressed
+  */
+  onPress?: () => void
```

Comment:
do we need this? it's already inherited from TouchableOpacityProps above. or you're just adding it so we get the custom ts doc comment?

Reply:
Adding this removes a red squiggly on `onPress` in the de-structuring of the props. 😄

Reply:
I think the red squiggly was there because you haven't run `yarn` in the boilerplate thus extending a blank type. once you install the deps, the squiggly disappears.

Reply:
before/after `yarn`

![yulolimum-capture-2022-08-15--16-23-41@2x](https://user-images.githubusercontent.com/1775841/184740509-229c4406-0915-483e-b983-2c0314bee9b0.jpg)

Reply:
Ah, I didn't even think of that. This might be sunk cost fallacy at work, but I think I'd prefer to keep `onPress` specifically called out here, because we use that to switch between `TouchableOpacity` and `View` for the icon wrapper. Thoughts? Low GAFO here, I can revert this change if you feel strongly about it.

Reply:
I'm cool with keeping it bro! In which case, let's type it as `onPress?: TouchableOpacityProps["onPress"]` since the function actually has arguments.